app/controllers/api/v1/icons_controller.rb

```diff
@@ -0,0 +1,27 @@
+class Api::V1::IconsController < Api::ApiController
+  before_filter :login_required, only: :destroy
+
+  resource_description do
+    description 'Processes S3 uploads that were not used'
+  end
+
+  api! 'Given an S3 key that the user has not turned into an icon, deletes the file from S3'
+  param :s3_key, String, required: true, desc: 'S3 object key'
+  error 401, "You must be logged in"
+  error 403, "Icon does not belong to the user"
+  error 422, "Invalid parameters provided: s3 key is in use."
+  def s3_delete
+    unless params[:s3_key].starts_with?("users/#{current_user.id}")
```

Comment:
can we check for the trailing slash too, so user 3 can't delete user 31's (unused but uploaded) icons? (probably not a major problem but still)